sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_models.py

```diff
@@ -216,57 +245,77 @@ def _from_generated(cls, definition):
 
 class _Operation(object):
     def __init__(self, **kwargs):
-        self.status = kwargs.get("status", None)
-        self.status_details = kwargs.get("status_details", None)
-        self.error = kwargs.get("error", None)
-        self.start_time = kwargs.get("start_time", None)
-        self.end_time = kwargs.get("end_time", None)
-        self.job_id = kwargs.get("job_id", None)
+        self._status = kwargs.get("status", None)
+        self._status_details = kwargs.get("status_details", None)
+        self._start_time = kwargs.get("start_time", None)
+        self._end_time = kwargs.get("end_time", None)
+        self._job_id = kwargs.get("job_id", None)
+
+        error = kwargs.get("error", None)
+        # responses can return an empty error object when there's no error
+        if error is None or error.code is None:
+            self._error = None
+        else:
+            self._error = AzureError("Error: {}".format(error.message or self._status_details), error=error)
```

Comment:
It's odd to use an exception type as a data object, and AzureError has an awkward API for this case. It won't document the structure of `error`, which it would expose as its `inner_exception`, which is misleading because `error` isn't an exception. Have you considered [ODataV4Format](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/core/azure-core/azure/core/exceptions.py#L105)?

Reply:
I did notice ODataV4Format, and I think that would be a better choice -- I just need to figure out how to provide the generated Error object we get back as a JSON object that ODataV4Format can understand